yetanotherco/aligned_layer
# feat: configurable reconnection attempts on web socket

Added config `batcher.eth_ws_reconnects` config param as maximum reconnection attempts.

> Known issue: if the web socket connection gets an error (e.g Connection refused) it won't retry its reconnection

# To test

1. Start anvil
2. Start batcher
3. Kill anvil
4. You should see a message like:
```
[2024-06-03T21:47:09Z ERROR ethers_providers::rpc::transports::ws::backend] Error response from WS err=WebSocket protocol error: Connection reset without closing handshake
[2024-06-03T21:47:09Z INFO  ethers_providers::rpc::transports::ws::manager] Reconnecting to backend remaining=99999999999998 url="ws://localhost:8545"
```

Note that im creating PR anyways, because in prod it reached max reconnection attempts, so it was not getting error connection refused
